### ISSUE TITLE
`#[graphql(skip)]` does not works on complex fields

### ISSUE THREAD

**Issue by IcanDivideBy0** [graphql-rust/juniper#482]
**Describe the bug**
When using `#[graphql(skip)]` on a "complex" field (a struct fn) the field is still avaliable through the GraphQL endpoint.

**To Reproduce**
```rust
pub struct Foo {
    pub foo: i32,
    pub bar: i32,
}

#[juniper::object]
impl Foo {
    pub fn foo(&self) -> i32 {
        self.foo
    }

    #[graphql(skip)]
    pub fn bar(&self) -> i32 { // This field can still be accessed through the API
        self.bar
    }
}
```

**Expected behavior**
`#[graphql(skip)]` should works on both "simple" and "complex" fields.

**Additional context**
N/A

**Comment by davidpdrsn** [comment]
Would you expect that `bar` would become a method on `Foo`? If so that isn't quite how `juniper::object` works. [Attribute macros](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) are able to "erase" the input code and instead insert something completely different. That is what `juniper::object` does. It generates an implementation of [GraphQLType](https://docs.rs/juniper/0.14.1/juniper/trait.GraphQLType.html) and doesn't keep the `impl` block that you wrote. So `foo` and `bar` don't become methods on `Foo`.

Long story short, but if you want to be able to call `Foo::bar` you need to add another `impl Foo` that isn't annotated with `#[juniper::object]` and define the method there.

**Comment by LegNeato** [comment]
We would love a PR explaining this in the book!

**Comment by IcanDivideBy0** [comment]
Ok, thanks for the answer. It feels a bit weird to write twice the `impl` block, but it works well this way.
